crypto/fipsmodule/ecdh/ecdh.c

```diff
@@ -79,26 +79,47 @@
 
 int ECDH_compute_shared_secret(uint8_t *buf, size_t *buflen, const EC_POINT *pub_key,
```

Comment:
We should insert a service indicator lock/unlock in this function. `EC_KEY_check_fips` is an approved function and will inadvertently update the state when called here.

Reply:
Only if the context has an associate private key no?

Reply:
Oh yeah, that's true. Didn't look at the condition closely enough.